src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/ComponentBroadcaster.cs

```diff
@@ -182,56 +182,56 @@ private void EnsureComponentInitialized()
             }
         }
 
-        protected virtual void SendComponentCreation(IEnumerable<SocketEndpoint> newConnections)
+        protected virtual void SendComponentCreation(IEnumerable<INetworkConnection> newConnections)
         {
-            if (endpointsNeedingComponentCreation != null)
+            if (connectionsNeedingComponentCreation != null)
             {
-                for (int i = endpointsNeedingComponentCreation.Count - 1; i >= 0; i--)
+                for (int i = connectionsNeedingComponentCreation.Count - 1; i >= 0; i--)
                 {
-                    if (ShouldSendChanges(endpointsNeedingComponentCreation[i]))
+                    if (ShouldSendChanges(connectionsNeedingComponentCreation[i]))
                     {
-                        SendComponentCreation(endpointsNeedingComponentCreation[i]);
-                        endpointsNeedingComponentCreation.RemoveAt(i);
+                        SendComponentCreation(connectionsNeedingComponentCreation[i]);
+                        connectionsNeedingComponentCreation.RemoveAt(i);
                     }
                 }
             }

```

Comment:
Can we make it a bit more idiomatic? Something like:
`connectionsNeedingComponentCreation?.RemoveAll(x => SendComponentCreationIfShould(x));`
where `bool SendComponentCreationIfShould(...)` is also directly reusable below.

Reply:
Opted for TrySendComponentCreation